watchtower-plugin/tests/test.py

```diff
@@ -133,28 +127,17 @@ def test_retry_watchtower(node_factory, bitcoind, teosd):
     # Make a new payment with an unreachable tower
     l1.rpc.pay(l2.rpc.invoice(25000000, "lbl1", "desc1")["bolt11"])
 
-    # The retrier manager waits 1 second before spawning new retriers for unreachable towers,
-    # so we need to wait a little bit until a retrier is started for our tower.
-    while l2.rpc.gettowerinfo(tower_id)["status"] == "temporary_unreachable":
-        time.sleep(1)
+    # The retrier manager waits 1 second before spawning new retriers for unreachable towers.
+    # Wait for the "Setting <tower_id> as unreachable log so we are sure the tower is unreachable"
+    l2.daemon.wait_for_log("as unreachable")
     assert l2.rpc.gettowerinfo(tower_id)["status"] == "unreachable"
     assert l2.rpc.gettowerinfo(tower_id)["pending_appointments"]
 
     # Start the tower and retry it
     teosd.start()
 
-    # Even though we set the max retry time to zero seconds, the retrier manager takes some time (1s) to recognize
-    # that the tower is unreachable. So manual retries might fail as the tower is marked as temporary unreachable.
-    while True:
-        try:
-            l2.rpc.retrytower(tower_id)
-            break
-        except Exception:
-            time.sleep(1)
```

Comment:
I think this part is needed to be retried multiple time.
This has to do with the retrier manager polling the towers to retry every time delta (I need to review how the retrier logic works and confirm).

Reply:
There shouldn't be an issue with this anymore because the tower was flagged as unreachable a couple of LOC earlier

Reply:
This is indeed true, I think we had this in a while loop cuz towers were being set to unreachable but the retrier manager wasn't throwing their retriers away yet. I reviewed the retrier manager logic and it seems it throws away retriers right after it sets the tower to be unreachable in the same retry round. So the sleep is no issue here indeed.

Reply:
Yeah, that may have been from before the retrier was reworked because some race conditions were posible (I think this was #89)